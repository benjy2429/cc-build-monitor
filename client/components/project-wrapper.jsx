import React from 'react';
import Summary from './summary';
import projectSelector from '../selectors/project-selector';

class ProjectWrapper extends React.PureComponent {
  render() {
    const { building, passing, failing, unknown } = this.props.projects;

    return (
      <div className="project-wrapper">
        <Summary projects={building} category="building" />
        <Summary projects={failing} category="failing" />
        <Summary projects={unknown} category="unknown" />
        <Summary projects={passing} category="passing" />
      </div>
    );
  }
}

export default projectSelector(ProjectWrapper);
export { ProjectWrapper as UnwrappedProjectWrapper };
