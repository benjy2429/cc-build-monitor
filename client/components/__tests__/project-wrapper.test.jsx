import React from 'react';
import { shallow } from 'enzyme';
import { UnwrappedProjectWrapper as ProjectWrapper } from '../project-wrapper';

const defaultProps = {
  projects: {
    building: [{ name: 'org/project-1' }],
    passing: [{ name: 'org/project-2' }],
    failing: [{ name: 'org/project-3' }],
    unknown: [{ name: 'org/project-4' }],
  },
};

describe('Project wrapper', () => {
  it('renders correctly', () => {
    const component = shallow(
      <ProjectWrapper {...defaultProps} />,
    );

    expect(component).toMatchSnapshot();
  });
});
