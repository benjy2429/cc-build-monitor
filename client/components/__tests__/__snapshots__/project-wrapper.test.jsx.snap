// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project wrapper renders correctly 1`] = `
<div
  className="project-wrapper"
>
  <Summary
    category="building"
    projects={
      Array [
        Object {
          "name": "org/project-1",
        },
      ]
    }
  />
  <Summary
    category="failing"
    projects={
      Array [
        Object {
          "name": "org/project-3",
        },
      ]
    }
  />
  <Summary
    category="unknown"
    projects={
      Array [
        Object {
          "name": "org/project-4",
        },
      ]
    }
  />
  <Summary
    category="passing"
    projects={
      Array [
        Object {
          "name": "org/project-2",
        },
      ]
    }
  />
</div>
`;
